database/src/types.rs

```diff
@@ -49,96 +45,290 @@ use bytemuck::{AnyBitPattern, NoUninit, Pod, Zeroable};
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
-//---------------------------------------------------------------------------------------------------- TestType
-/// TEST
+//---------------------------------------------------------------------------------------------------- Aliases
+// TODO: document these, why they exist, and their purpose.
+//
+// TODO: should we go as far as making transparent
+// wrappers for all of these to be more typesafe?
```

Comment:
Stuff like `Amount`, `KeyImage` are just type aliases for now.

I think we should do something like `KeyImage(pub [u8; 32])` instead? Returning newtypes would be a lot more type safe.

Reply:
I don't think this is something to worry about too much, if we are adding data to the wrong tables I think we would find out pretty quick right?